code/datums/components/_component.dm

```diff
@@ -310,18 +312,16 @@
 /datum/proc/_SendSignal(sigtype, list/arguments)
 	var/target = comp_lookup[sigtype]
 	if(!length(target))
-		var/datum/C = target
-		if(!(C.datum_flags & DF_SIGNAL_ENABLED))
+		var/datum/listening_datum = target
+		if(!(listening_datum.datum_flags & DF_SIGNAL_ENABLED))
 			return NONE
-		var/proctype = C.signal_procs[src][sigtype]
-		return NONE | CallAsync(C, proctype, arguments)
+		return NONE | CallAsync(listening_datum, listening_datum.signal_procs[src][sigtype], arguments)
```

Comment:
this might not be important enough to justify cutting down on the readability, it depends on how much we care about more people being able to understand the internals of _SendSignal() i suppose